web/src/main/java/org/springframework/security/web/firewall/StrictHttpFirewall.java

```diff
@@ -134,13 +135,13 @@ public class StrictHttpFirewall implements HttpFirewall {
 
 	private static final Predicate<String> HEADER_VALUE_PREDICATE = (s) -> HEADER_VALUE_PATTERN.matcher(s).matches();
 
-	private Predicate<String> allowedHeaderNames = ASSIGNED_AND_NOT_ISO_CONTROL_PREDICATE;
```

Comment:
Let's keep both. There is the public constant that folks can use to build expressions and then there is whatever in the internal setting. You'd want something like this:

```java
private Predicate<String> allowedHeaderNames = ALLOWED_HEADER_NAMES;

public static final Predicate<String> ALLOWED_HEADER_NAMES = ASSIGNED_AND_NOT_ISO_CONTROL_PREDICATE;
```

Reply:
Thank you for your feedback, @jzheaux.

I have updated the code to reflect your suggestions. 

One question I have is regarding the potential confusion that might arise because ALLOWED_HEADER_NAMES is final and thus immutable. Even if the user changes the allowedHeaderNames predicate, the ALLOWED_HEADER_NAMES constant remains unchanged. Could this lead to any misunderstandings or misuses?

Reply:
This is a common standard in Java, so I don't think it will lead to wide confusion. The constant needs to be final because that is not the thing that is being changed. Instead, the setter is updating the value for `allowedHeaderNames`.